contrib/epee/src/file_compression.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2014-2021, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "file_compression.h"
+
+#ifdef WIN32
+    #include <boost/iostreams/filter/zlib.hpp>
+    #define COMPRESSOR zlib_compressor()
+    #define DECOMPRESSOR zlib_decompressor()
+    #define EXTENSION ".zip"
+#else 
+    #include <boost/iostreams/filter/gzip.hpp>
+    #define COMPRESSOR gzip_compressor()
+    #define DECOMPRESSOR gzip_decompressor()
+    #define EXTENSION ".gz"
+#endif // WIN32
```

Comment:
So this uses a different compression format for windows and others ?

Reply:
Yes. For Windows it uses just ZIP, so it's easier for the users to unpack yhe archives with the OS supported functionality (Windows Explorer). 

This concerns files only. For eventual inter OS comms, I will of course have to use GZ everywhere.